### PR TITLE
brotab: init at 0.0.5

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Browser tab controller - https://github.com/balta2ar/brotab

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/misc/brotab/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, glibcLocales, python3 }:
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]:
```suggestion
{ lib, fetchFromGitHub, glibcLocales, python3 }:
```

File: pkgs/tools/misc/brotab/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    requests
+    flask
+    requests
+    pytest
+    psutil
+  ];
+
+  meta = {
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]: personal pref
```suggestion
  meta = with lib; {
```

File: pkgs/tools/misc/brotab/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    homepage = "https://github.com/balta2ar/brotab";
+    description = "Control your browser's tabs from the command line";
+    license = stdenv.lib.licenses.mit;
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]:
```suggestion
    license = licenses.mit;
```

File: pkgs/tools/misc/brotab/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    maintainers = with stdenv.lib.maintainers; [ doronbehar ];
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]:
```suggestion
    maintainers = with maintainers; [ doronbehar ];
```

File: pkgs/tools/misc/brotab/default.nix (hunk 1 above)
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]: I think python as a variable is easier to conceptualize. Just pass the version in all-packages.nix
```suggestion
{ stdenv, fetchFromGitHub, glibcLocales, python }:
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -18130,6 +18130,8 @@ in
 
   browsh = callPackage ../applications/networking/browsers/browsh { };
 
+  brotab = callPackage ../tools/misc/brotab { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: related to other change :)
```suggestion
  brotab = callPackage ../tools/misc/brotab {
    python = python3;
  };
```

**Comment by doronbehar** [comment]
Thanks for the comments @jonringer !

File: pkgs/tools/misc/brotab/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{ lib, fetchFromGitHub, glibcLocales, python }:
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]: unused
```suggestion
{ lib, fetchFromGitHub, python }:
```

**Review by jonringer — COMMENTED** [review]
this repo seems to be most a 1 person show with no other [repositories currently packaging](https://repology.org/projects/?search=brotab&maintainer=&category=&inrepo=&notinrepo=&repos=&families=&repos_newest=&families_newest=) this.

Why should ilt be added to nixpkgs?

**Comment by doronbehar** [comment]
> Why should ilt be added to nixpkgs?

Well first of all, the fact it's not famous enough so that other repos will include it doesn't mean the package is not useful. The least I _can_ in order to support this software is to distribute it, starting with NixOS.

Besides that, it's not just a script that one can `git clone` and link to his `$PATH` - the package consists of several executables the browser extension needs to be aware of. Hence it's almost impossible for NixOS users to use it without installing it the Nix way.

**Comment by doronbehar** [comment]
What say you @jonringer ?

File: pkgs/tools/misc/brotab/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  propagatedBuildInputs = with python.pkgs; [
+    requests
+    flask
+    requests
+    pytest
+    psutil
+  ];
+
```
Comment (jonringer) [inline pkgs/tools/misc/brotab/default.nix]: I would still like tests, this is what I came up with for the unit tests:

```suggestion

  # test_integration.py requires Chrome browser session
  checkPhase = ''
    ${python.interpreter} -m unittest brotab/tests/test_{brotab,utils}.py
  '';
  
```

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
has tests :+1: 

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73777
1 package built:
brotab
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build brotab

**Comment by JohnAZoidberg** [comment]
This is actually a pretty cool tool!
Let's see how helpful it is for managing my 500+ windows :P

**Comment by doronbehar** [comment]
Thanks! :smile:

**Pull request merged**